torch/csrc/jit/passes/dead_code_elimination.cpp

```diff
@@ -36,10 +36,47 @@ bool hasSideEffects(Node * node, bool_memo_type& memo) {
   return has_side_effects;
 }
 
+void removeDeadIfOutputs(Node* node) {
+  if (node->kind() != prim::If)
+    return;
+  for(size_t i_1 = node->outputs().size(); i_1 > 0; --i_1) {
+    size_t i = i_1 - 1;
+    if (!node->outputs().at(i)->hasUses()) {
+      node->eraseOutput(i);
+      for (Block* b : node->blocks()) {
+        b->eraseOutput(i);
+      }
+    }
+  }
+}
+
+void removeDeadLoopOutputs(Node* node) {
+  if (node->kind() != prim::Loop)
+    return;
+  auto loop_body = node->blocks().at(0);
+  auto loop_input_offset = 2; // offset of loop carried deps in input list
+  auto loop_body_offset = 1; // offset to the loop carried dependencies in block inputs/outputs
```

Comment:
constexpr?

Reply:
I am fixing this in the next PR with ir_views.h which will get rid of all the weird loop indexing logic.